packages/material-ui/src/StepButton/StepButton.js

```diff
@@ -10,14 +10,17 @@ export const styles = {
   /* Styles applied to the root element. */
   root: {
     width: '100%',
-    padding: '24px 16px',
-    margin: '-24px -16px',
     boxSizing: 'content-box',
   },
   /* Styles applied to the root element if `orientation="horizontal"`. */
-  horizontal: {},
+  horizontal: {
```

Comment:
This change remembers me https://github.com/mui-org/material-ui/pull/15872/files#diff-4d2a2d052bcab32af90fcb671b5ec5f6R16. I believe it's a breaking change for people using `theme.overrides.MuiStepButton.root`. They will no longer override the padding and margin. They have to target `horizontal` instead of `root`. What do you think of only changing the styles of the `vertical` key?

Reply:
I'll take a look. Really not liking the indirection for default styles

Reply:
Yeah really unfortunate that this was placed in the `root`.

Reply:
The problem is that there's no good way to say now The initial margin/padding styles but with the following changed values. I would have to say margin: 0; padding 0 8 but then I might override other user defined styles.

Reply:
Yeah I would just ice it as a breaking change and wait to v5. IMO overriding the spacing in the `root` class is already a bug in app code because it will create the same issue we currently have.

Reply:
What do you think of?:
```diff
diff --git a/packages/material-ui/src/StepButton/StepButton.js b/packages/material-ui/src/StepButton/StepButton.js
index 7572b0bb2..accda6f61 100644
--- a/packages/material-ui/src/StepButton/StepButton.js
+++ b/packages/material-ui/src/StepButton/StepButton.js
@@ -15,7 +15,10 @@ export const styles = {
     boxSizing: 'content-box',
   },
   /* Styles applied to the root element if `orientation="horizontal"`. */
-  horizontal: {},
+  horizontal: {
+    margin: 0,
+    padding: '8px 0',
+  },
   /* Styles applied to the root element if `orientation="vertical"`. */
   vertical: {
     justifyContent: 'flex-start',
```

Reply:
> Could you expand on that? I don't see the issue.

They're repeating the same bug we currently create. The margin should not be applied universally but depending on the variant. The variants don't have a common spacing

>> Really not liking the indirection for default styles

> What does it meant?

To find out what the margin for the horizontal variant is one has to know that this is the default variant and then look in the root styling as opposed to just looking into horizontal stylings.

The variant applies different spacing behavior as opposed to e.g. color. horizontal or vertical has no impact on color which means that it could be shared in the root class. 

The problem is illustrated in this issue. I don't want the padding from the `root` styling but have no way of disabling it without using `overrides` from the theme. 

If there is a way of disabling the root padding for the vertical variant I'm happy to apply it here. But it seems like I can only explicitly set it which is not the same.

Reply:
Interesting, would this work? I would expect the output to be correct and to have the overrides working as before.

```jsx
export const styles = {
  /* Styles applied to the root element. */
  root: {
    width: '100%',
    boxSizing: 'content-box',
    padding: '24px 16px',
    margin: '-24px -16px',
  },
  /* Styles applied to the root element if `orientation="horizontal"`. */
  horizontal: {},
  /* Styles applied to the root element if `orientation="vertical"`. */
  vertical: {
    padding: '8px 0',
    margin: 0,
    justifyContent: 'flex-start',
  },
  /* Styles applied to the `ButtonBase` touch-ripple. */
  touchRipple: {
    color: 'rgba(0, 0, 0, 0.3)',
  },
};
```

Reply:
@eps1lon What do you think?